### PR TITLE
docs(api): add documentation for tap and ap

### PR DESCRIPTION
**Pull request by TylorS**

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -171,20 +171,45 @@ Replace each event value with x.::
...
+
 ap
-^^
+^^^
 
 .. code-block:: haskell
 
   ap :: Stream (a -> b) -> Stream a -> Stream b
 
+Apply the latest function in a stream of functions to the latest value in stream.
```
Comment (briancavalier) [inline docs/api.rst]: The end of the sentence, "latest value in stream", seems like a copypasta artifact.  I wonder if it works to try to make the second half parallel to the first, like:

"Apply the latest function in a stream of functions to the latest value in a stream of values."

Does that read ok?

File: docs/api.rst [hunk 2]
```diff
@@ -171,20 +171,45 @@ Replace each event value with x.::
...
+
+.. code-block:: javascript
+
+  streamOfFunctions:              --f---------g--------h------>
+  stream:                         -a-------b-------c-------d-->
+  ap(stream, streamOfFunctions.): --fa-----fb-gb---gc--hc--hd->
```
Comment (briancavalier) [inline docs/api.rst]: This one is really hard to draw an ascii diagram for.  Do you think the `fa`, `fb`, etc stuff will make sense to readers?  I know it's what's in the most.js docs, but should we try to come up with something better, or do you think it's ok?

File: docs/api.rst (hunk 1 above)
Comment (Frikki) [inline docs/api.rst]: Does it matter if it is a "stream of values"? I mean, couldn’t it be a stream of anything? How about:

"Apply the latest emitted function in a stream of functions to the latest emitted value of another stream."

File: docs/api.rst (hunk 2 above)
Comment (Frikki) [inline docs/api.rst]:
```
streamOfFunctions:              --f-----------g---------h--------->
stream:                         -a-------b---------c---------d---->
ap(stream, streamOfFunctions.): --f(a)---f(b)-g(b)-g(c)-h(c)-h(d)->
```

File: docs/api.rst (hunk 1 above)
Comment (briancavalier) [inline docs/api.rst]: @frikki I like it :+1:

We've been trending away from the word emitted, but imho, your suggestion still sounds good after omitting it:

"Apply the latest function in a stream of functions to the latest value of another stream."

**Pull request merged**